src/main/webapp/js/instructorFeedbackEdit/customFeedbackPaths.js

```diff
@@ -1,368 +1,379 @@
+    FEEDBACK_PARTICIPANT_TYPE_INSTRUCTORS: 'INSTRUCTORS',
+    FEEDBACK_PARTICIPANT_TYPE_TEAMS: 'TEAMS',
+    FEEDBACK_PARTICIPANT_TYPE_OWN_TEAM: 'OWN_TEAM',
+    FEEDBACK_PARTICIPANT_TYPE_OWN_TEAM_MEMBERS: 'OWN_TEAM_MEMBERS',
+    FEEDBACK_PARTICIPANT_TYPE_OWN_TEAM_MEMBERS_INCLUDING_SELF: 'OWN_TEAM_MEMBERS_INCLUDING_SELF',
+    FEEDBACK_PARTICIPANT_TYPE_NONE: 'NONE',
+    TEAM_NAME_INSTRUCTORS: 'Instructors',
 
-$(document).ready(function() {
-    initializeCustomFeedbackPathsData();
-    initializeFeedbackPathsSpreadsheets();
-    bindEventHandlers();
-});
+    // These variables are read-only and are not changed after they have been initialized
+    // They are used to derive the data that is used to fill the spreadsheets
+    sessionCreator: null,
+    studentEmailToTeamNameMap: null,
+    instructorEmails: null,
+    teamNameToStudentEmailsMap: null,
+    studentEmails: null,
+    teamNames: null,
+    allPossibleFeedbackGivers: null,
+    allPossibleFeedbackRecipients: null,
 
-function initializeCustomFeedbackPathsData() {
-    sessionCreator = $('#session-creator-data').data('session-creator');
-    studentEmailToTeamNameMap = $('#students-data').data('students');
-    instructorEmails = $('#instructors-data').data('instructors');
-    teamNameToStudentEmailsMap = [[]];
-    studentEmails = [];
-    teamNames = [];
-    
-    for (var studentEmail in studentEmailToTeamNameMap) {
-        if (studentEmailToTeamNameMap.hasOwnProperty(studentEmail)) {
-            studentEmails.push(studentEmail);
-            
-            var teamName = studentEmailToTeamNameMap[studentEmail];
-            if (!teamNames.includes(teamName)) {
-                teamNames.push(teamName);
+    initializeCustomFeedbackPathsData: function() {
+        customFeedbackPaths.sessionCreator = $('#session-creator-data').data('session-creator');
+        customFeedbackPaths.studentEmailToTeamNameMap = $('#students-data').data('students');
+        customFeedbackPaths.instructorEmails = $('#instructors-data').data('instructors');
+        customFeedbackPaths.teamNameToStudentEmailsMap = [[]];
+        customFeedbackPaths.studentEmails = [];
+        customFeedbackPaths.teamNames = [];
+        
+        for (var studentEmail in customFeedbackPaths.studentEmailToTeamNameMap) {
+            if (customFeedbackPaths.studentEmailToTeamNameMap.hasOwnProperty(studentEmail)) {
+                customFeedbackPaths.studentEmails.push(studentEmail);
+                
+                var teamName = customFeedbackPaths.studentEmailToTeamNameMap[studentEmail];
+                if (!customFeedbackPaths.teamNames.includes(teamName)) {
+                    customFeedbackPaths.teamNames.push(teamName);
+                }
+                
+                var studentEmailsList = customFeedbackPaths.teamNameToStudentEmailsMap[teamName];
+                if (studentEmailsList === undefined) {
+                    studentEmailsList = [];
+                }
```

Comment:
Willl `studentEmailsList = studentEmailslist || []` work? If so, which is the more common way to initialize?